### PR TITLE
cleanup(storage): handle cases in emulator where metadata is optional

### LINKED ISSUE #6692: Storage(emulator): Handle requests with empty data
There are cases where the request sent to the emulator has empty data. `json.loads(request.data)` causes an `JSONDecodeError` when `request.data` is an empty byte string. One example in the emulator is [`objects.copy()`](https://github.com/googleapis/google-cloud-cpp/blob/591fa09e2009e1c4b855e513c26457d037195004/google/cloud/storage/emulator/emulator.py#L564) where metadata can be optionally supplied.

### PR DESCRIPTION
**Pull request by cojenco**
Revise storage emulator to handle cases where metadata can be optionally supplied in request body in [`storage.objects.copy`](https://cloud.google.com/storage/docs/json_api/v1/objects/copy#request-body) and [`storage.objects.rewrite`](https://cloud.google.com/storage/docs/json_api/v1/objects/rewrite#request-body)

Fixes #6692

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6777)

### CODE REVIEW

File: google/cloud/storage/emulator/emulator.py [hunk 1]
```diff
@@ -561,7 +561,8 @@ def objects_copy(src_bucket_name, src_object_name, dst_bucket_name, dst_object_n
         flask.request, dst_metadata, dst_media, dst_bucket, True, None, dst_rest_only
     )
     db.insert_object(flask.request, dst_bucket_name, dst_object, None)
-    dst_object.patch(flask.request, None)
+    if len(flask.request.data):
```
Comment (dbolduc) [inline google/cloud/storage/emulator/emulator.py:564]: would `if flask.request.data:` be more pythonic?

File: google/cloud/storage/emulator/emulator.py [hunk 2]
```diff
@@ -625,7 +626,8 @@ def objects_rewrite(src_bucket_name, src_object_name, dst_bucket_name, dst_objec
             dst_rest_only,
         )
         db.insert_object(flask.request, dst_bucket_name, dst_object, None)
-        dst_object.patch(rewrite.request, None)
+        if len(flask.request.data):
```
Comment (dbolduc) [inline google/cloud/storage/emulator/emulator.py:629]: same here

File: google/cloud/storage/emulator/emulator.py (hunk 1 above)
Comment (cojenco) [reply google/cloud/storage/emulator/emulator.py]: yes it would, thank you :)

**Review by dbolduc — APPROVED** [review]

**Comment by dbolduc** [comment]
Thanks for the fix!

**Pull request merged**